keras_nlp/models/roberta.py

```diff
@@ -170,6 +171,184 @@ def get_config(self):
+    ):
+        super().__init__(**kwargs)
+        self.sequence_length = sequence_length
+        if truncate not in ("round_robin", "waterfall"):
+            raise ValueError(
+                "Only 'round_robin' and 'waterfall' algorithms are "
+                "supported. Received %s" % truncate
+            )
+        self.truncate = truncate
+        self.start_value = start_value
+        self.end_value = end_value
+        self.pad_value = pad_value
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "sequence_length": self.sequence_length,
+                "start_value": self.start_value,
+                "end_value": self.end_value,
+                "pad_value": self.pad_value,
+                "truncate": self.truncate,
+            }
+        )
+        return config
+
+    def _sanitize_inputs(self, inputs):
+        """Force inputs to a list of rank 2 ragged tensors."""
+        # Sanitize inputs.
+        if not isinstance(inputs, (list, tuple)):
+            inputs = [inputs]
+        if not inputs:
+            raise ValueError("At least one input is required for packing")
+        input_ranks = [x.shape.rank for x in inputs]
+        if not all(0 < rank < 3 for rank in input_ranks):
+            raise ValueError(
+                "All inputs for packing must have rank 1 or 2. "
+                f"Received input ranks: {input_ranks}"
+            )
+        if None in input_ranks or len(set(input_ranks)) > 1:
+            raise ValueError(
+                "All inputs for packing must have the same rank. "
+                f"Received input ranks: {input_ranks}"
+            )
+        return inputs
+
+    def _convert_dense(self, x):
+        """Converts inputs to rank 2 ragged tensors."""
+        if isinstance(x, tf.Tensor):
+            return tf.RaggedTensor.from_tensor(x)
+        else:
+            return x
+
+    def _trim_inputs(self, inputs):
+        """Trim inputs to desired length."""
+        # Special tokens include the start token at the beginning of the
+        # sequence, two `end_value` at the end of every segment save the last,
+        # and the `end_value` at the end of the sequence.
+        num_special_tokens = 2 * len(inputs)
+        if self.truncate == "round_robin":
+            return tf_text.RoundRobinTrimmer(
+                self.sequence_length - num_special_tokens
+            ).trim(inputs)
+        elif self.truncate == "waterfall":
+            return tf_text.WaterfallTrimmer(
+                self.sequence_length - num_special_tokens
+            ).trim(inputs)
+        else:
+            raise ValueError("Unsupported truncate: %s" % self.truncate)
+
+    def _combine_inputs(self, segments):
+        """Combine inputs with start and end values added."""
+        dtype = segments[0].dtype
+        batch_size = segments[0].nrows()
+
+        start_value = tf.convert_to_tensor(self.start_value, dtype=dtype)
```

Comment:
seems like this could look simpler, something like

```
start_value = tf.convert_to_tensor(self.start_value, dtype=dtype)
end_value = tf.convert_to_tensor(self.end_value, dtype=dtype)
start_col = tf.fill((batch_size, 1), start_value)
end_col = tf.fill((batch_size, 1), end_value)

segments_to_combine = []
for i, seg in enumerate(segments):
  segments_to_combine.append(start_col if i == 0 else end_col)
  segments_to_combine.append(seg)
  segments_to_combine.append(end_col)
```